src/Functions/array/arrayFill.cpp

```diff
@@ -128,8 +128,59 @@ using FunctionArrayReverseFill = FunctionArrayMapped<ArrayFillImpl<true>, NameAr
 
 REGISTER_FUNCTION(ArrayFill)
 {
-    factory.registerFunction<FunctionArrayFill>();
-    factory.registerFunction<FunctionArrayReverseFill>();
+    FunctionDocumentation::Description description = R"(
+The `arrayFill` function sequentially processes a source array from the first element
+to the last, evaluating a lambda condition at each position using elements from
+the source and condition arrays. When the condition evaluates to false at
+position i, the function replaces that element with the element at position i-1
+from the current state of the array. The first element is always preserved
+regardless of any condition.
+
+`arrayFill` is a [higher-order function](/sql-reference/functions/overview#higher-order-functions). You must pass a lambda function to it as the first argument, and it can't be omitted.
+)";
+    FunctionDocumentation::Syntax syntax = "arrayFill(λ(x [, y1, ..., yN]), source [, cond1, ... , condN])";
+    FunctionDocumentation::Arguments arguments = {
+        {"λ(x [, y1, ..., yN])", "A lambda function `λ(x [, y1, y2, ... yN]) → F(x [, y1, y2, ... yN])` which operates on elements of the source array (`x`) and condition arrays (`y`). [Lambda function](/sql-reference/functions/overview#arrow-operator-and-lambda)."},
```

Comment:
I got a bit creative here with `λ(x [, y1, ..., yN])`. Let me know if it is too creative...

Reply:
Creativity is good, consistency is better :-) Let's use plain old `func`.